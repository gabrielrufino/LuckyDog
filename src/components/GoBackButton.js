import React from 'react';
import Icon from 'react-native-vector-icons/FontAwesome5';
import {TouchableOpacity} from 'react-native';

export default function GoBackButton({onPress}) {
  return (
    <TouchableOpacity onPress={onPress}>
      <Icon size={25} name="arrow-left" color="#C8AB83" />
    </TouchableOpacity>
  );
}
