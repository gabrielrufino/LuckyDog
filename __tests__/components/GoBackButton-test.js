import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import GoBackButton from '../../src/components/GoBackButton';

describe('components/GoBackButton.js', () => {
  it('Should renders the go back button correctly', () => {
    const onPress = jest.fn();

    const tree = renderer.create(<GoBackButton onPress={onPress} />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
