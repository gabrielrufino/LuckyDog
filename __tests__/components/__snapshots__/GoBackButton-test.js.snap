// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/GoBackButton.js Should renders the go back button correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "color": "#C8AB83",
          "fontSize": 25,
        },
        undefined,
        Object {
          "fontFamily": "FontAwesome5Free-Solid",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {
          "fontWeight": "900",
        },
      ]
    }
  >
    ÔÅ†
  </Text>
</View>
`;
